File: TypeScript/0209-长度最小的子数组.ts
function minSubArrayLen(target: number, nums: number[]): number {
    let left: number = 0,
        right: number = 0,
        sum: number = 0;

    const n: number = nums.length;
    let result: number = n + 1;

    while (right < n) {
        sum += nums[right];
        if (sum >= target) {
            while (sum - nums[left] >= target) {
                sum -= nums[left++];
            }
            result = Math.min(result, right - left + 1);
        }
        right++;
    }
    return result === n + 1 ? 0 : result;
}

File: Java/0209-长度最小的子数组.java
class Solution {
  public int minSubArrayLen(int target, int[] nums) {
    int sum = nums[0];
    if (sum >= target) return 1;
    int left = 0, right = 0, n = nums.length, result = Integer.MAX_VALUE;
    while (++right < n) {
      sum += nums[right];
      while (sum >= target && left <= right) {
        result = Math.min(result, right - left + 1);
        sum -= nums[left++];
      }
    }
    return result == Integer.MAX_VALUE ? 0 : result;
  }
}
